openedx_ledger/views.py

```diff
@@ -2,5 +2,96 @@
 Views for the openedx_ledger app.
 """
 # from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views.generic import View
 
-# Create your views here.
+from openedx_ledger.api import reverse_full_transaction
+from openedx_ledger.models import Transaction
+from openedx_ledger.signals.signals import ADMIN_INVOKED_TRANSACTION_UNENROLL
+
+
+# TODO: Should we add a base class to do auth checks?
+class UnenrollLearnersView(View):
+    """
+    Admin view for the canceling platform enrollments form.
+    """
+    template = "edx_ledger/admin/unenroll.html"
+
+    def get(self, request, transaction_id):
+        """
+        Handle GET request - render "Cancel transaction without refund" form.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+            transaction_uuid (str): Enterprise Customer UUID
+
+        Returns:
+            django.http.response.HttpResponse: HttpResponse
+        """
+        return render(
+            request,
+            self.template,
+            {'transaction': Transaction.objects.get(uuid=transaction_id)}
+        )
+
+    def post(self, request, transaction_id):
+        """
+        Handle POST request - handle form submissions.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+        """
+        print("sending admin invoked transaction unenroll signal")
+        ADMIN_INVOKED_TRANSACTION_UNENROLL.send(
+            sender=self,
+            transaction=Transaction.objects.get(uuid=transaction_id),
+            admin=request.user,
+        )
+        url = reverse("admin:openedx_ledger_transaction_change", args=(transaction_id,))
+        return HttpResponseRedirect(url)
+
+
+class UnenrollLearnersWithRefundView(View):
+    """
+    Admin view for the canceling platform enrollments and reversing transaction objects form.
+    """
+    template = "edx_ledger/admin/unenroll_with_refund.html"
+
+    def get(self, request, transaction_id):
+        """
+        Handle GET request - render "Cancel transaction with refund" form.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+            transaction_uuid (str): Enterprise Customer UUID
+
+        Returns:
+            django.http.response.HttpResponse: HttpResponse
+        """
+        return render(
+            request,
+            self.template,
+            {'transaction': Transaction.objects.get(uuid=transaction_id)}
+        )
+
+    def post(self, request, transaction_id):
+        """
+        Handle POST request - handle form submissions.
+        """
+        print("Reversing transaction and sending admin invoked transaction unenroll signal")
+        transaction = Transaction.objects.get(uuid=transaction_id)
+        # TODO: determine how to generate idempotency key
```

Comment:
[curious] Is this still a relevant TODO for this PR or is it getting tackled elsewhere?

Reply:
That's there to remind me to discuss this with @johnnagro before this gets merged 😄